libs/librepcb/project/boards/drc/boarddesignrulecheck.cpp

```diff
@@ -405,18 +405,31 @@ void BoardDesignRuleCheck::checkMinimumPthRestring(int progressStart,
           library::FootprintPad::BoardSide::THT) {
         continue;  // skip SMT pads
       }
-      PositiveLength size =
-          qMin(pad->getLibPad().getWidth(), pad->getLibPad().getHeight());
-      Length restring = (*size - *pad->getLibPad().getDrillDiameter() + 1) / 2;
-      if (restring < *mOptions.minPthRestring) {
-        QString msg = QString(tr("Min. pad restring ('%1'): %2",
-                                 "Placeholders are pad name + restring width"))
-                          .arg(pad->getDisplayText().simplified(),
-                               formatLength(restring));
-        PositiveLength diameter =
-            PositiveLength(pad->getLibPad().getDrillDiameter() + 1) +
-            mOptions.minPthRestring + mOptions.minPthRestring;
-        Path location = Path::circle(diameter).translated(pad->getPosition());
+      Length width       = *pad->getLibPad().getWidth();
+      Length height      = *pad->getLibPad().getHeight();
+      Length drillWidth  = 0;
+      Length drillHeight = 0;
+      if (pad->getLibPad().getDrillSize()) {
+        drillWidth  = *pad->getLibPad().getDrillSize()->getWidth();
+        drillHeight = *pad->getLibPad().getDrillSize()->getHeight();
+      }
+      Length widthRestring  = (width - drillWidth + 1) / 2;
+      Length heightRestring = (height - drillHeight + 1) / 2;
+      if (widthRestring < *mOptions.minPthRestring or
+          heightRestring < *mOptions.minPthRestring) {
+        QString msg =
+            QString(tr("Min. pad restring ('%1'): %2",
+                       "Placeholders are pad name + restring width"))
+                .arg(pad->getDisplayText().simplified(),
+                     formatLength(qMin(widthRestring, heightRestring)));
+        PositiveLength locWidth = PositiveLength(
+            drillWidth + *mOptions.minPthRestring + *mOptions.minPthRestring);
+        PositiveLength locHeight = PositiveLength(
+            drillHeight + *mOptions.minPthRestring + *mOptions.minPthRestring);
+        Path location =
+            Path::obround(locWidth, locHeight)
+                .translated(pad->getPosition())
+                .rotated(pad->getRotation() + device->getRotation());
```

Comment:
In some cases, a wrong position is shown for this kind of DRC message:

![grafik](https://user-images.githubusercontent.com/5374821/85755633-69c27c00-b70e-11ea-9de4-ff1c90074f2a.png)

I suspect you have to do the rotation first, and then the translation(?). And probably mirroring also needs to be considered.

Or as an alternative, just revert it to the original implementation which only shows a circle instead of an oblong :wink:

Reply:
oh yes, I'm guessing .rotated rotates around the origin, makes sense :), will fix